src/Microsoft.ML.Transforms/NAIndicatorTransform.cs

```diff
@@ -68,116 +78,196 @@ private static VersionInfo GetVersionInfo()
         internal const string FriendlyName = "NA Indicator Transform";
         internal const string ShortName = "NAInd";
 
-        private static string TestType(ColumnType type)
+        internal static string TestType(ColumnType type)
         {
             // Item type must have an NA value. We'll get the predicate again later when we're ready to use it.
             Delegate del;
             if (Conversions.Instance.TryGetIsNAPredicate(type.ItemType, out del))
                 return null;
             return string.Format("Type '{0}' is not supported by {1} since it doesn't have an NA value",
-                type, LoaderSignature);
+                type, LoadName);
+        }
+
+        public class ColumnInfo
+        {
+            public readonly string Input;
+            public readonly string Output;
+
+            /// <summary>
+            /// Describes how the transformer handles one column pair.
+            /// </summary>
+            /// <param name="input">Name of input column.</param>
+            /// <param name="output">Name of output column.</param>
+            public ColumnInfo(string input, string output)
+            {
+                Input = input;
+                Output = output;
+            }
         }
 
-        private const string RegistrationName = "NaIndicator";
+        private const string RegistrationName = nameof(NAIndicatorTransform);
```

Comment:
>nameof(NAIndicatorTransform); [](start = 48, length = 29)

The usage of nameof here is great! Shall we get rid of the LoadName and just substitute with this?

Reply:
Sounds good! 

---
In reply to: [224316082](https://github.com/dotnet/machinelearning/pull/1217#discussion_r224316082) [](ancestors = 224316082)

Reply:
I disagree because of the above comment

---
In reply to: [224647220](https://github.com/dotnet/machinelearning/pull/1217#discussion_r224647220) [](ancestors = 224647220,224316082)